libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/Bubble.kt

```diff
@@ -0,0 +1,122 @@
+package net.cyclestreets.views.overlay
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.*
+import android.net.Uri
+import android.view.MotionEvent
+import android.webkit.URLUtil
+import androidx.core.content.ContextCompat
+import net.cyclestreets.util.Draw
+import org.osmdroid.api.IProjection
+import org.osmdroid.views.MapView
+import net.cyclestreets.Undoable
+import net.cyclestreets.view.R
+import net.cyclestreets.views.CycleMapView
+import org.osmdroid.views.Projection
+
+object Bubble {
```

Comment:
For future reference, it would be helpful to reviewers if you add a comment on significant refactors like this to say "This is X extracted from Y" or whatever.  I just figured that out myself.

Reply:
Where's the best place to put that comment?
I mentioned it the summary in the comment on the PR.
Oh wait, sorry, it's in the original PR.  I should have copied over the summary and attachments.

Reply:
As a general point I always treat refactorings of existing code separate to other commits that add new functionality, even if the refactoring is needed because of the addition of that new functionality. This is because the refactoring is a good improvement in itself, and can standalone irrespective of functionality changes.

Using `git add -p` to do partial commits is very helpful for this.

Reply:
I tend to agree @mvl22, it's a good practice. Makes things more digestible to review also!

Reply:
Yes, it was a lot to plough through!  I was hoping my summary notes would make it a bit more digestible :-)